apps/src/blockly/addons/functionEditor.js

```diff
@@ -101,10 +101,10 @@ export default class FunctionEditor {
     // we have to pass the main ws so that the correct procedures are populated
     // false to not show the new function button inside the modal editor
     this.editorWorkspace.registerToolboxCategoryCallback('PROCEDURE', () =>
-      functionsFlyoutCategory(Blockly.mainBlockSpace, true)
```

Comment:
why was this changed?

Reply:
+1 It looks like passing the main workspace here vs. the editor workspace was intentional: `we have to pass the main ws so that the correct procedures are populated`. But we did move away from the `mainBlockSpace` language?

Reply:
We need to register this category using the editor workspace so that when we `getNewFunctionButtonWithCallback`, it register a button callback for the editor workspace. Otherwise, the button doesn't due anything if it's in the flyout for the modal editor. Note that we also register these category callbacks on the main workspace in the wrapper. We no longer have to look at the main workspace to find the procedures, so I'll removed that comment the Emily is quoting above.

Reply:
So we don't care if the "new function"/"new behavior" button is rendered inside the editor toolbox? I see it in your screenshots and I don't think that's how it worked before (and I imagine, if we could avoid that, we'd want to?).

Reply:
![Screenshot 2024-01-17 at 4 25 37 PM](https://github.com/code-dot-org/code-dot-org/assets/26844240/692a460f-c64b-4200-ac00-7bdfed678305)
![Screenshot 2024-01-17 at 4 25 58 PM](https://github.com/code-dot-org/code-dot-org/assets/26844240/d5007f15-3de3-448a-baa0-f7e9ca1cdb44)

```
We need to register this category using the editor workspace so that when we getNewFunctionButtonWithCallback, it register a button callback for the editor workspace. Otherwise, the button doesn't due anything if it's in the flyout for the modal editor.
``` 
But you're saying we want this button? I thought we a) didn't want it and b) didn't want it to do anything. (Sorry for two comments, my 4:30 PM brain is dragging.)

Reply:
Do you recall a reason why we wouldn't to be able to create a function/behavior from within the editor? I don't think there's a technical reason we can't, so maybe it's worth checking with Amy?

On CDO Blockly, you are right that you can't create a new function if the function editor is open (same for behaviors). Strangely, you _can_ create a function if just the behavior editor is open (and vice versa). If you try this, things get gross, because CDO makes two modals which it isn't really meant to. :( 

I'm not saying I know for sure, but I wonder if the decision to hide these buttons in the original modal editor was a technical one or a design decision.

Reply:
> We no longer have to look at the main workspace to find the procedures
Why is this?

Reply:
@mikeharv I think it was a design decision, but I'm not sure it's one I disagree with?

Reply:
@molly-moen More detailed answer in Slack, but we always populate the flyout by looking for procedure blocks on both the main and hidden workspaces. I think the comment has long been out of date.

@ebeastlake I've made a partial revert so that we again only add the "Create" buttons to the flyout if we're on the main workspace. I've left in the change that would make these buttons work correctly if we did want to use them in the modal editor's toolbox at any point.